### PR TITLE
Fix 24073 - fix founding date validation

### PR DESCRIPTION
**Pull request by ArwenQin**
*Issue #:* /bcgov/entity#24073

*Description of changes:*
- Fix the bug due to different date format

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/services/filings/validations/continuation_in.py [hunk 1]
```diff
@@ -192,7 +192,21 @@ def validate_business_in_colin(filing_json: dict, filing_type: str) -> list:
...
-        elif founding_date != response_json['business']['foundingDate']:
-            msg.append({'error': 'Founding date does not match with founding date from Colin.',
-                        'path': business_founding_date_path})
+        else:
+            # Check if founding date is None
+            if founding_date is None:
+                msg.append({
+                    'error': 'Founding date is missing in the filing data.',
+                    'path': business_founding_date_path
+                })
```
Comment (ArwenQin) [inline legal-api/src/legal_api/services/filings/validations/continuation_in.py:201]: Do we need this?

File: legal-api/src/legal_api/services/filings/validations/continuation_in.py [hunk 2]
```diff
@@ -192,7 +192,21 @@ def validate_business_in_colin(filing_json: dict, filing_type: str) -> list:
...
+            else:
+                # Truncate both dates to "YYYY-MM-DDTHH:MM:SS" for comparison
+                founding_date_truncated = founding_date[:19]
+                colin_founding_date_truncated = response_json['business']['foundingDate'][:19]
+
```
Comment (ArwenQin) [inline legal-api/src/legal_api/services/filings/validations/continuation_in.py:206]: Colin is  '2009-07-23T07:00:00-00:00', the filing json is '2009-07-23T07:00:00.000+00:00'
truncated after 19th

File: legal-api/src/legal_api/services/filings/validations/continuation_in.py (hunk 1 above)
Comment (severinbeauvais) [reply legal-api/src/legal_api/services/filings/validations/continuation_in.py:201]: If the schema already makes this required then no.

**Review by severinbeauvais — APPROVED** [review]
LGTM.

Vysakh, please merge if OK with you.

File: legal-api/src/legal_api/services/filings/validations/continuation_in.py (hunk 1 above)
Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/services/filings/validations/continuation_in.py:201]: its not a required filed as per schema

Comment (ArwenQin) [reply legal-api/src/legal_api/services/filings/validations/continuation_in.py:201]: Just to confirm how to handle this: if per filing json is none, do we check Colin returns none too (does this happen?); or return error message?

File: legal-api/src/legal_api/services/filings/validations/continuation_in.py (hunk 2 above)
Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/services/filings/validations/continuation_in.py:206]: This is not a good approach. Can you try this one
```suggestion
datetime.fromisoformat(founding_date) !=  datetime.fromisoformat(response_json['business']['foundingDate'])
```

Comment (ArwenQin) [reply legal-api/src/legal_api/services/filings/validations/continuation_in.py]: Updated, thanks. Only concern is whether Colin can have different time zone? Other than -00:00, like +03:00. If not I think this approach is safe

**Review by vysakh-menon-aot — APPROVED** [review]

**Pull request merged**